### ISSUE TITLE
NodeJS Static Domain example

### ISSUE THREAD

**Issue by GradySain** [ngrok/ngrok-nodejs#83]
I've tried a number of diff approaches to get a static domain working in node. Currently stuck at:

```
const express = require("express");
const ngrok = require("@ngrok/ngrok");
const app = express();

app.get("/", (req, res) => {
  res.send("Hello World!");
});

async function setup() {
  // create session
  const session = await new ngrok.NgrokSessionBuilder()
    .authtokenFromEnv()
    .metadata("Online in One Line")
    .connect();
  // create tunnel
  const tunnel = await session
    .httpEndpoint()
    .allowCidr("0.0.0.0/0")
    .oauth("google")
    // .requestHeader("X-Req-Yup", "true")
    .listen();
  // link tunnel to app
  const socket = await ngrok.listen(app, tunnel);
  console.log(`Ingress established at: ${tunnel.url()}`);
  console.log(`Express listening on: ${socket.address()}`);
}

setup();
```

getting the error: `[Error: failed to start tunnel: Your account is not authorized to use ip restrictions.` I commented out the requestHeader stuff because it threw errors too...

**Comment by nijikokun** [comment]
Thanks for trying out the Node SDK @GradySain! 🎉 

If you comment out the following line:

```
    .allowCidr("0.0.0.0/0")
```

Does that work or return an error?

**Comment by GradySain** [comment]
Thanks for the fast response. I'm sure I'm doing tons of stuff wrong.

I commented out allowCider and oauth (it threw an error)...

The app launched, I got a tunnel `https://2ae9-75-7-99-127.ngrok.io`, clicked it, got an error:

`ERR_NGROK_3004
ngrok gateway error The server returned an invalid or incomplete HTTP response.`

I should have noted, I'm trying to understand the new free static domain usage, and get it working using this `ngrok-nodejs` package. I'm basically new to ngrok (have used it in the past). 

My cursory understanding, start a new node session, this starts a new random tunnel, but that tunnel can always be accessed via free static domain?

 thanks!

**Comment by GradySain** [comment]
Should have noted also, I have my Authtoken in .env file like this:

NGROK_AUTHTOKEN= xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

but i'm not doing anything like `require('dotenv').config();`, it seems like this is handled via `.authtokenFromEnv()`? DOn't know if Authtoken should be a string in quotes?

**Comment by CK-Ward** [comment]
Hi @GradySain, try exporting your authtoken in your environment, it might not be picking it up from the env file. Mine's configured in my .zsh_profile without quotes :)

**Comment by nijikokun** [comment]
> it might not be picking it up from the env file
> I have my Authtoken in .env file like this

Yep, right now it only pulls from the `process.env` variable and not the `.env` file, if you want to do that you'll have to use something like `dotenv`. Maybe something we can support in the future!

**Comment by GradySain** [comment]
hmm, I'm still floundering...

```
require("dotenv").config();
console.log(process.env.NGROK_AUTHTOKEN);

const express = require("express");
const ngrok = require("@ngrok/ngrok");
const app = express();

app.get("/", (req, res) => {
  res.send("Hello World!");
});

async function setup() {
  // create session
  const session = await new ngrok.NgrokSessionBuilder().authtokenFromEnv().metadata("Online in One Line").connect();
  // create tunnel
  const tunnel = await session
    .httpEndpoint()
    // .allowCidr("0.0.0.0/0")
    // .oauth("google")
    // .requestHeader("X-Req-Yup", "true")
    .listen();
  // link tunnel to app
  const socket = await ngrok.listen(app, tunnel);
  console.log(`Ingress established at: ${tunnel.url()}`);
  console.log(`Express listening on: ${socket.address()}`);
}

setup();
```

I get: `Ingress established at: https://d5e9-75-7-99-127.ngrok-free.app`

Click that link, get: `ERR_NGROK_3004 ngrok gateway error The server returned an invalid or incomplete HTTP response.`

My node app errors (crashes):

```
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^
[Error: error forwarding: Os { code: 2, kind: NotFound, message: "The system cannot find the file specified." }] {
  code: 'GenericFailure'
}
```

I tried passing the token as an arg to .authtokenFromEnv(), and tried the token as a string, same results...

**Comment by GradySain** [comment]
If it helps, I'm on windows 10 ... ?

**Comment by nijikokun** [comment]
@GradySain I replicated your entire environment and I get the same error. Going to hand this over to the engineering team, and I will try to debug a little myself.

OS: Windows 11
ENV: .env (`dotenv`)
npm: `10.1.0`
nodejs: `v20.5.0`
shell: `gitbash mingw`

![image](https://github.com/ngrok/ngrok-nodejs/assets/198276/3544a77c-c7d7-480a-a768-9ee36220fcd1)
![image](https://github.com/ngrok/ngrok-nodejs/assets/198276/49b4f9eb-4675-45f4-ac47-160f96ec1528)

**Comment by GradySain** [comment]
@nijikokun thanks very much, good to know I'm not completely crazy. I'll monitor and hopefully it's something simple :)

**Comment by bobzilladev** [comment]
Hi @GradySain,

The `system cannot find the file specified` issue should now be fixed in version `0.7.1` released earlier today.

**Comment by bobzilladev** [comment]
I'll close this ticket out, feel free to open a new one if you run into any other issues!
